src/pint/models/timing_model.py

```diff
@@ -322,11 +322,11 @@ def free_params(self):
 
     @free_params.setter
     def free_params(self, params):
-        params = {self.match_param_aliases(p) for p in params}
+        params_true = {self.match_param_aliases(p) for p in params}
         for p in self.params:
-            getattr(self, p).frozen = p not in params
-            params.discard(p)
-        if params:
+            getattr(self, p).frozen = p not in params_true
+            params_true.discard(p)
+        if params_true:
             raise ValueError(
                 "Parameter(s) are familiar but not in the model: {}".format(params)
```

Comment:
Is there any example for this case? Shouldn't the params_true be a subset of self.params?

Reply:
I *think* this branch will never be reached, because aliases should not be translated for parameters that are not actually in the model. But in case someone should change the implementation of `match_param_aliases` (for example if aliases were tracked at the Component level instead of the Parameter level), I think it's good to have this here to ensure an exception is raised.

Reply:
I agree to have a function in the component level is a good idea. But that can be a separate issue.